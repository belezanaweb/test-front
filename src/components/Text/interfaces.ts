import { ReactNode } from 'react'

export interface TextStyledProps {
  size?: 'small' | 'medium' | 'large' | 'xlarge' | 'huge'
  weight?: 'regular' | 'medium' | 'bold'
  $color?: 'black' | 'grey' | 'primary' | 'grey-weak' | 'grey-strong' | 'danger' | 'grey-default' | 'white'
  fullWidth?: boolean
  decoration?: string
}

export interface TextProps extends TextStyledProps {
  children: ReactNode
  as?: 'h1' | 'p' | 'span'
}
