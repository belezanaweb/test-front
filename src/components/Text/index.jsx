import React from 'react';
import PropTypes from 'prop-types';

import TextStyled from './style';

const Text = props => (
  <TextStyled
    data-cy={props.dataCy && props.dataCy}
    size={props.size && props.size}
    bold={props.bold && props.bold}
    color={props.color && props.color}
    center={props.center && props.center}
    uppercase={props.uppercase && props.uppercase}
    className={props.className && props.className}
  >
    {props.children && props.children}
  </TextStyled>
);

Text.propTypes = {
  size: PropTypes.string,
  bold: PropTypes.bool,
  color: PropTypes.string,
  center: PropTypes.bool,
  uppercase: PropTypes.bool,
};

export default Text;
