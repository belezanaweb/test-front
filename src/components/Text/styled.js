import styled from 'styled-components'
import { colors, fonts } from '../../assets/styled/tokens'

export const StyledText = styled.p`
  height: 17px;
  width: 100%;
  color: ${colors.dustyGrayText};
  font-family: ${fonts.text};
  font-size: 14px;
  font-weight: 700;
  line-height: 17px;
  margin-left: 20px;
`
