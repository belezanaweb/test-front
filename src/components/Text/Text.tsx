import styled from 'styled-components';

const Text = styled.span`
  color: ${({ theme }) => theme.colors.grayDark};
  display: inline-flex;
  font-family: ${props => props.theme.fontFamily};
  font-size: 14px;
  line-height: 16px;
  letter-spacing: -0.015em;
`;

export default Text;
