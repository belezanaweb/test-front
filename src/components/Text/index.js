import React from 'react'
import { StyledText } from './styled'

const Text = (props) => {
  return (
    <>
      <StyledText>{props.text}</StyledText>
    </>
  )
}

export { Text }
