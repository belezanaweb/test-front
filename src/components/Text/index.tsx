import { TextProps } from './interfaces'
import * as Styled from './styled'

const Text = ({ children, size, weight, $color, decoration, fullWidth, as = 'span' }: TextProps) => (
  <Styled.Tag as={as} size={size} weight={weight} $color={$color} fullWidth={fullWidth} decoration={decoration}>
    {children}
  </Styled.Tag>
)

export default Text
