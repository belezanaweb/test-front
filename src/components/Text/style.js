import styled from 'styled-components'; // eslint-disable-line import/no-extraneous-dependencies

const TextStyled = styled.p`
  font-size: ${props => (props.size ? props.size : '13px')};
  color: ${props => (props.color ? props.color : 'black')};
  text-align: ${props => (props.center ? 'center' : 'left')};
  text-transform: ${props => (props.uppercase ? 'uppercase' : 'initial')};
  letter-spacing: 0.93px;
  font-weight: ${props => (props.bold ? 'bold' : 'normal')};
  font-family: 'HelveticaNeueCyr';
  margin: ${props => (props.margin ? props.margin : '0')};
  font-style: normal;
`;

export default TextStyled;
