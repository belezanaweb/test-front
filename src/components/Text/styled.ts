import styled, { css } from 'styled-components'
import { TextStyledProps } from './interfaces'

export const Tag = styled.span<TextStyledProps>`
  width: ${({ fullWidth }) => (fullWidth ? '100%' : 'unset')};
  color: ${({ theme, $color }) => theme.color[$color] || theme.color.black};
  font-weight: ${({ theme, weight }) => weight || theme.font.weight.regular};
  font-size: ${({ theme, size }) => theme.font.size[size] || theme.font.size.small};

  ${({ decoration }) =>
    decoration &&
    css`
      text-decoration: ${decoration};
    `}
`
