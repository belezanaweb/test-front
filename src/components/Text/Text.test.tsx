import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import { provider } from '../../utils/testUtils';

import Text from './Text';

describe('<Text />', () => {
  it('should render Text', () => {
    const { container } = provider(<Text>The quick brown fox jumps over the lazy dog</Text>, render);

    expect(container).toMatchSnapshot();
  });
});
