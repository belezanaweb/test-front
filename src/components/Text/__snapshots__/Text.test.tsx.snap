// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Text /> should render Text 1`] = `
<div>
  <span
    class="sc-AxjAm kPvYly"
  >
    The quick brown fox jumps over the lazy dog
  </span>
</div>
`;
